### PR TITLE
fix String#lastIndexOf for non ASCII strings

### LINKED ISSUE #120: String.prototype.lastIndexOf broken
String.prototype.lastIndexOf is broken when passed with the Unicode characters and second argument specified.

```javascript
'abcあい'.lastIndexOf('あい', 1)
```

expected: -1
actual: 3

According to [ECMA 21.1.3.9](http://www.ecma-international.org/ecma-262/6.0/#sec-string.prototype.lastindexof) - the `position` argument is used for the end of the string to look *left* from - ie, it is indeed spec-compliant behavior for `'abcあい'.lastIndexOf('あい', n)` to return `-1` when `n` is less than `'abcあい'.indexOf('あい')`.

### PR DESCRIPTION
**Pull request by sankha93**
Fixes #120

### CODE REVIEW

**Comment by dilijev** [comment]
Nice! LGTM

File: test/Strings/lastindexof.js [hunk 1]
```diff
@@ -27,6 +27,9 @@ WScript.Echo(str2.lastIndexOf("\0ab"));
 var str3 = "\u0100\u0111\u0112\u0113";
 WScript.Echo(str3.lastIndexOf("\u0112\u0113"));
 
+var str4 = "\u0061\u0062\u0063\u3042\u3044";
+WScript.Echo(str4.lastIndexOf("\u3042\u3044", 1));
```
Comment (goyakin) [inline test/Strings/lastindexof.js]: It doesn't look like we have existing test coverage for this case and, therefore, I think we should add more tests. The ones I have in mind are:

  * `str4.lastIndexOf("\u3042\u3044") === 3`: Successful match without `position`
  * `str4.lastIndexOf("\u3042\u3044", 4) === 3`: Successful match ending at `position`
  * `"\u3042\u3044\u0061".lastIndexOf("\u3042\u3044", 3) === 0`: Successful match ending before `position`
  * `str4.lastIndexOf("\u3044\0", 5) === -1`: Verify that we don't match `"\0"` at the end of the string in C++

**Comment by abchatra** [comment]
This change looks good to me. 

Not related to this change. Though I am a bit surprised that outer loop is not bound by searchLen. It would a tad faster in failing cases.

File: test/Strings/lastindexof.js (hunk 1 above)
Comment (sankha93) [inline test/Strings/lastindexof.js]: @goyakin I have updated the tests to include these cases.

**Comment by sankha93** [comment]
@abchatra I have updated the outer loop to incorporate `searchLen`.

**Comment by abchatra** [comment]
@sankha93 pretty cool! This change looks good to me.

**Comment by goyakin** [comment]
Thanks @sankha93 for contributing!

**Pull request merged**